hibernate-envers/src/test/java/org/hibernate/envers/test/integration/manytoone/lazy/BaseDomainEntityVersion.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.envers.test.integration.manytoone.lazy;
+
+import java.util.Objects;
+
+import javax.persistence.Column;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+
+/**
+ * @author Chris Cranford
+ */
+@MappedSuperclass
+public abstract class BaseDomainEntityVersion extends BaseDomainEntityMetadata {
+    private static final long serialVersionUID = 1564895954324242368L;
+
+    @Id
+    @Column(name = "version", nullable = false, updatable = false)
+    private long version;
+
+    public long getVersion() {
+        return version;
+    }
+
+    public void setVersion(long version) {
+        this.version = version;
+    }
+
+    public abstract Object getId();
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        BaseDomainEntityVersion that = (BaseDomainEntityVersion) o;
+        return Objects.equals(getId(), that.getId()) && version == that.version;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getId(), version);
+    }
```

Comment:
Again, not formatted using space char delimiters. Maybe not a big deal for testing code?

Reply:
Hi @NathanQingyangXu, thanks for the review.  Just for clarity, the Hibernate build doesn't apply the checkstyle enforcement to test code for a variety of reasons.  I'll fix these mainly for clarity but just so you know for the future.  In general, if a code style violation happens, the build should fail to pass the checks below.

Reply:
Thanks for letting me know testing code is out of scope of checkstyle. As a newbie, I review mainly from generic coding perspective without approval, unless I happened to be certain I am qualified to make decision. I am learning fast and hopefully I might contribute to full review soon.